tools/export/cmake/__init__.py

```diff
@@ -74,6 +75,12 @@ def generate(self):
         # sort includes reverse, so the deepest dir comes first (ensures short includes)
         includes = sorted([re.sub(r'^[.]/', '', l) for l in self.resources.inc_dirs], reverse=True)
 
+        # select dependant hex files to merge into compiled hex image
+        hex_files = self.resources.hex_files
+        if hasattr(self.toolchain.target, 'hex_filename'):
+            hex_filename = self.toolchain.target.hex_filename
+            hex_files = list(f for f in hex_files if basename(f) == hex_filename)
```

Comment:
This is the same thing:

```suggestion
            hex_files = [f for f in hex_files if basename(f) == hex_filename]
```

Reply:
Thanks for suggestion, will use the syntax next time I touch the exporter code.
For now I'd like to keep the code in cmake and makefile exporters in sync.

Reply:
Should it be a function on the base exporter class then?

Reply:
I can add new property to the base Exporter class, similar to existing [libraries ](https://github.com/ARMmbed/mbed-os/blob/mbed-os-5.11.4/tools/export/exporters.py#L122)property, and update cmake and makefile exporters to use it. Please confirm this is correct:
```python
@property
def hex_files(self):
    hex_files = self.resources.hex_files
    if hasattr(self.toolchain.target, 'hex_filename'):
        hex_filename = self.toolchain.target.hex_filename
        hex_files = [f for f in hex_files if basename(f) == hex_filename]
    return hex_files
```

If so, I will update this PR with the proposed change and update both exporters to use it.

Reply:
I pushed https://github.com/ARMmbed/mbed-os/pull/9756/commits/cf0ea921046c4c4eb8102f7e9cc72dec86ec6242 and tested it has no functional impact (verified `mbed export -m CY8CKIT_062_BLE -i cmake_gcc_arm` and `mbed export -m CY8CKIT_062_BLE -i eclipse_gcc_arm`).

I also used [] to initalize the python list.

Please review. I updated the PR description to mention this is [X] Refactor